src/components/Pagination/Pagination.tsx

```diff
@@ -2,0 +1,96 @@
+import cn from 'classnames';
+import { PageGhangeCallback } from '../../types/PageChangeCallback';
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange(page: number | PageGhangeCallback): void;
+}
+
+const getPages = (total: number, perPage: number) => {
+  const pages = [];
+
+  for (let i = 1; i <= Math.ceil(total / perPage); i++) {
+    pages.push(i);
+  }
+
+  return pages;
+};
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pages = getPages(total, perPage);
+
+  const handlePrevNextPage = (increase: boolean) => {
+    if (currentPage !== 1 && !increase) {
+      return onPageChange((currentState => {
```

Comment:
The casting to 'PageGhangeCallback' is unnecessary. Since 'onPageChange' can accept a number or a callback, TypeScript will infer the correct type. Remove the casting to simplify the code.